lib/ansible/galaxy/collection/__init__.py

```diff
@@ -58,495 +129,93 @@
-        return info
-
-    @staticmethod
-    def collection_info(b_path, fallback_metadata=False):
-        info = CollectionRequirement.artifact_info(b_path)
-        if info or not fallback_metadata:
-            return info
-        return CollectionRequirement.galaxy_metadata(b_path)
-
-    def add_requirement(self, parent, requirement):
-        self.required_by.append((parent, requirement))
-        new_versions = set(v for v in self.versions if self._meets_requirements(v, requirement, parent))
-        if len(new_versions) == 0:
-            if self.skip:
-                force_flag = '--force-with-deps' if parent else '--force'
-                version = self.latest_version if self.latest_version != '*' else 'unknown'
-                msg = "Cannot meet requirement %s:%s as it is already installed at version '%s'. Use %s to overwrite" \
-                      % (to_text(self), requirement, version, force_flag)
-                raise AnsibleError(msg)
-            elif parent is None:
-                msg = "Cannot meet requirement %s for dependency %s" % (requirement, to_text(self))
-            else:
-                msg = "Cannot meet dependency requirement '%s:%s' for collection %s" \
-                      % (to_text(self), requirement, parent)
+def verify_local_collection(
+        local_collection, remote_collection,
+        artifacts_manager,
+):  # type: (Candidate, Candidate, ConcreteArtifactsManager) -> None
+    """Verify integrity of the locally installed collection.
 
-            collection_source = to_text(self.b_path, nonstring='passthru') or self.api.api_server
-            req_by = "\n".join(
-                "\t%s - '%s:%s'" % (to_text(p) if p else 'base', to_text(self), r)
-                for p, r in self.required_by
-            )
+    :param local_collection: Collection being checked.
+    :param remote_collection: Correct collection.
+    :param artifacts_manager: Artifacts manager.
+    """
+    b_temp_tar_path = (  # NOTE: AnsibleError is raised on URLError
+        artifacts_manager.get_artifact_path
+        if remote_collection.is_concrete_artifact
+        else artifacts_manager.get_galaxy_artifact_path
+    )(remote_collection)
+
+    b_collection_path = to_bytes(
+        local_collection.src, errors='surrogate_or_strict',
+    )
 
-            versions = ", ".join(sorted(self.versions, key=SemanticVersion))
-            if not self.versions and self.pre_releases:
-                pre_release_msg = (
-                    '\nThis collection only contains pre-releases. Utilize `--pre` to install pre-releases, or '
-                    'explicitly provide the pre-release version.'
-                )
-            else:
-                pre_release_msg = ''
+    display.vvv("Verifying '{coll!s}'.".format(coll=local_collection))
+    display.vvv(
+        "Installed collection found at '{path!s}'".
+        format(path=to_native(local_collection.src)),
```

Comment:
Display should use text strings, will need a to_text in the format and u prefix the string so we don't have any ASCII encoding errors on Python 2.